src/content/github/file/main.tsx

```diff
@@ -151,34 +187,36 @@ async function process(metadata: FileMetadata): Promise<void> {
       previousElement: globals.coverageButton!,
       onClick: handleComponentClick,
       selectedOptions: selectedComponents,
-    })
-      .then(({ button, list }) => {
-        globals.componentsButton = button;
-        globals.componentsDrop = new Drop({
-          target: button,
-          content: list,
-          classes: "drop-theme-arrows codecov-z1 codecov-bg-white",
-          position: "bottom right",
-          openOn: "click",
-        });
-      })
+    }).then(({ button, list }) => {
+      globals.componentsButton = button;
+      globals.componentsDrop = new Drop({
+        target: button,
+        content: list,
+        classes: "drop-theme-arrows codecov-z1 codecov-bg-white",
+        position: "bottom right",
+        openOn: "click",
+      });
+    });
   }
 
+  // If commit sha is defined use that, otherwise just branch name
+  const getReportFn = metadata.commit ? getCommitReport : getBranchReport;
+
   let coverageReportResponses: Array<FileCoverageReportResponse>;
   try {
     if (selectedFlags?.length > 0) {
       coverageReportResponses = await Promise.all(
-        selectedFlags.map((flag) => getCommitReport(metadata, flag, undefined))
+        selectedFlags.map((flag) => getReportFn(metadata, flag, undefined))
       );
     } else if (selectedComponents?.length > 0) {
       coverageReportResponses = await Promise.all(
         selectedComponents.map((component) =>
-          getCommitReport(metadata, undefined, component)
+          getReportFn(metadata, undefined, component)
         )
       );
     } else {
```

Comment:
Still learning the extension bit by bit, are you not allowed mixing flags + components?

Reply:
Ooo good catch actually. Seems like as of today you cannot lol. I'll add a case for both